### PR TITLE
feat: added page_load and allow_tracking events for gtm

### PR DESCRIPTION
**Pull request by shayan-deriv**
## changes
- added the logic to send `page_load` and `allow_tracking` events to gtm so that core-web-vitals and google-analytics get triggered on p2p

### CODE REVIEW

File: src/routes/AppContent/index.tsx [hunk 1]
```diff
@@ -70,6 +71,13 @@ const AppContent = () => {
         setActiveTab(getActiveTab(location.pathname));
     }, [location]);
 
+    useEffect(() => {
```
Comment (amam-deriv) [inline src/routes/AppContent/index.tsx:74]: [M] let's use ref here to track if we have allowed, its more lightweight and usable for this use case

**Review by niloofar-deriv — APPROVED** [review]

File: src/routes/Router.tsx [hunk 2]
```diff
@@ -36,6 +35,12 @@ const RouteWithSubRoutes = (route: TRoutesWithSubRoutes) => {
 };
 
 const Router: FC = () => {
+    const location = useLocation();
```
Comment (amam-deriv) [inline src/routes/Router.tsx:38]: can we make a new hook here useHandleRouteChange?

Comment (shayan-deriv) [reply src/routes/Router.tsx]: done. please check again. not sure for now we need to add more to it or not

File: src/routes/AppContent/index.tsx (hunk 1 above)
Comment (shayan-deriv) [reply src/routes/AppContent/index.tsx:74]: done 👍

File: src/hooks/custom-hooks/useHandleRouteChange.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { useEffect } from 'react';
+import { useLocation } from 'react-router-dom';
+
+const useHandleRouteChange = () => {
```
Comment (farrah-deriv) [inline src/hooks/custom-hooks/useHandleRouteChange.ts:4]: please add a description of this hook

**Review by matin-deriv — APPROVED** [review]

**Review by farrah-deriv — APPROVED** [review]

**Review by cakasuma — APPROVED** [review]

**Pull request merged**